Fixes human readable print for last google protobuf

Hsilgos opened the pull request:
Google seems changed namespace ages ago for protobuf. Strange that nobody fixed it yet.

Hsilgos commented:
I signed it!

gennadiycivil commented:
@Hsilgos Could you please rebase this with the latest master

Hsilgos commented:
> @Hsilgos Could you please rebase this with the latest master

It's already on latest master, not sure how can i force travis-ci to rerun build.

Hsilgos commented:
> @Hsilgos Could you please rebase this with the latest master

Done

gennadiycivil commented:
Thank you for rebasing. 
Could you please explain in greater details what you are trying to achieve here. Best would be to have a reproducible unit test.

Hsilgos commented:
> Thank you for rebasing.
> Could you please explain in greater details what you are trying to achieve here. Best would be to have a reproducible unit test.

Sure. Seems that a lot of time ago gmock supported nice printing for Google protobuf's structures [there](https://github.com/google/googletest/blob/0599a7b8410dc5cfdb477900b280475ae775d7f9/googletest/include/gtest/gtest-printers.h#L169). Then Google seems renamed namespace from ::proto2 to ::google::protobuf in their protobuf library and gmock stopped to support it. Template struct IsAProtocolMessage is used [there](https://github.com/google/googletest/blob/0599a7b8410dc5cfdb477900b280475ae775d7f9/googletest/include/gtest/gtest-printers.h#L233) to detect that type is protobuf message.
So i just added new namespace and base class and IsAProtocolMessage now is able to detect protobuf messages again.
I added simple unit test but i'm afraid that it's everything i can do without adding protobuf as dependency to gmock =)
